### DISCUSSION TITLE
I can't use audio analog or bluetooth

### DISCUSSION THREAD

**Discussion by akirataniakira** [vicwomg/pikaraoke#795] [General]
it is not possible, I already did the basic configuration of both analog and bluetooth, I want to project with hdmi but capture the sound in another way

**Comment by vicwomg** [comment]
By analog do you mean the built-in 3.5mm headphone jack? pikaraoke should output from both the headphone jack and HDMI simulataneously.

As for bluetooth, you could try changing line 343 in karaoke.py
**from**: output = "alsa:hw:0,0" if self.alsa_fix else "both"
**to**: output = "alsa" if self.alsa_fix else "alsa"

If that works let me know and I can add a command line option for bluetooth.

**Comment by akirataniakira** [comment]
I made the indicated change but it didn't work to play via bluetooth, any other tips?

**Comment by Hartebee5t** [comment]
This may be the same as my own configuration. I use HDMI for video only, and send audio via Bluetooth to a wireless mic base unit as a mixer.  I did need to mess around with the underlying Pi settings to get it to work properly. 
I'll see if I can dig out what my changes involved.

**Comment by Hartebee5t** [comment]
One thing worth trying. I've found that even though the Pi Bluetooth connection with the mixer/receiver is reported as complete on each Pi power-up, it doesn't get any audio until I manually select another Bluetooth device from the Pi desktop Bluetooth menu right-click dropdown, then re-select the mixer/receiver device. After doing that I can launch PiKaraoke and the audio works fine.
I've yet to figure out a proper fix for this, as this is preventing me from running PiKaraoke fully headless (as I need to interact with the desktop every power-up). 
EDIT: There was already a comment under an earlier issue (#17, since Closed) which mentioned this, but I reckon it's really a Pi setup issue rather than a PiKaraoke app issue.

**Comment by Hartebee5t** [comment]
> I made the indicated change but it didn't work to play via bluetooth, any other tips?

Something I just discovered tonight is that if it's in this "Bluetooth not really connected" condition, if I go to the PiKaraoke Home screen and alternate between the Play and Pause buttons while a track is (silently) running, I can get brief {much less than 1/4 second} snippets of sound from the Bluetooth speaker. 

This is just an observation that someone might recognise as a symptom or clue to some configuration anomaly. If you're also able to get this behaviour then it would suggest w share the same problem.

**Comment by akever76** [comment]
Has anyone tried to use a USB to 3.5mm mini jack dongle adapter to output music from RPi5 to a sound mixer?  I just upgraded from RPi3 and didn't realize I was losing my headphones output.  What settings will I need to change?  Or does the USB dongle just work Plug and Play?

**Comment by vicwomg** [comment]
Wow, just learning about Raspberry Pi 5 not having a 3.5mm out. What a poor
decision on their part.

I've not tried the USB to 3.5mm myself, but I don't see any reason why they
wouldn't work, but might take a bit of configuration on the OS to send the
audio there.

**Comment by akever76** [comment]
It does work and if you are using the Raspberry Pi OS desktop environment like me, you simply right-click the speaker icon in the taskbar and choose the USB audio option for speaker output.  Piece of cake.  I just got it all set up.  Upgrading from Pi3b...wow much faster and lets me use the newest version of pikaraoke.  Thanks
